platforms/ios/config.cmake

```diff
@@ -82,3 +90,78 @@ set_target_properties(TangramMap PROPERTIES
   XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++14"
   XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++"
 )
+
+### Configure static library build target.
+
+add_library(tangram-static STATIC
+  ${TANGRAM_FRAMEWORK_SOURCES}
+)
+
+target_link_libraries(tangram-static PRIVATE
+  tangram-core
+  sqlite3
+  # Frameworks: use quotes so "-framework X" is treated as a single linker flag.
+  "-framework CoreFoundation"
+  "-framework CoreGraphics"
+  "-framework CoreText"
+  "-framework GLKit"
+  "-framework OpenGLES"
+  "-framework UIKit"
+)
+
+target_include_directories(tangram-static PRIVATE
+  platforms/common
+)
+
+# To produce a distributable version of our iOS SDK as a static library, we
+# need to collect all the symbols that will be needed in a final, linked
+# executable. CMake normally makes these symbols available by propagating link
+# flags from dependencies, but users who aren't building the SDK from source
+# won't have the dependency libraries in a build tree. We solve this by pre-
+# linking all of the dependency libraries that we would normally pass as flags.
+
+# Here we manually list the library files for all of our compiled dependencies.
+# I haven't found a way to get a full, recursive library list from CMake
+# so this list will need to be updated when any new library dependencies are
+# added. Note the quotes: this is needed to not make it a "list", which CMake
+# delimits with semicolons. Xcode expects a space-delimited list.
+set(TANGRAM_STATIC_DEPENDENCIES "\
+  $<TARGET_FILE:tangram-core>
```

Comment:
If its available, we could use something like this: https://gitlab.kitware.com/cmake/cmake/issues/12435